plugins/by-name/image/default.nix

```diff
@@ -124,12 +125,9 @@ lib.nixvim.plugins.mkNeovimPlugin {
     ];
   };
 
-  extraConfig = cfg: {
-    # In theory, we could remove that if the user explicitly disables `downloadRemoteImages` for
-    # all integrations but shipping `curl` is not too heavy.
-    dependencies = {
-      curl.enable = lib.mkDefault true;
-      ueberzug.enable = lib.mkIf (cfg.settings.backend == "ueberzug") (lib.mkDefault true);
-    };
-  };
+  # In theory, we could remove that if the user explicitly disables `downloadRemoteImages` for
+  # all integrations but shipping `curl` is not too heavy.
+  dependencies = [
+    "curl"
+  ] ++ lib.optional (config.plugins.image.settings.backend == "ueberzug") "ueberzug";
```

Comment:
I think it may be simpler to keep the conditional dependency in extraConfig, where it benefits from access to the `cfg` binding.

The main purpose of this `dependencies` argument is to enable us to auto-add documentation in the future, which doesn't work well with conditional deps anyway...

Reply:
_I guess_ the API could be extended to support _string or attrset_, where the attrset could include additional info:


```nix
dependencies = [
  "curl"
  {
    name = "ueberzug";
    when = config.plugins.image.settings.backend == "ueberzug";
    whenText = lib.literalExpression ''
      config.plugins.image.settings.backend == "ueberzug"
    '';
  }
];
```